runtime/compiler/env/VMJ9.cpp

```diff
@@ -6768,20 +6768,22 @@ TR_J9VMBase::isInterfaceClass(TR_OpaqueClassBlock * clazzPointer)
    }
 
 bool
-TR_J9VMBase::isEnumClass(TR_OpaqueClassBlock * clazzPointer, TR_ResolvedMethod *method)
+TR_J9VMBase::isEnumClass(TR_OpaqueClassBlock * clazzPointer)
    {
-   int32_t modifiersForClass = 0;
-   if (isClassArray(clazzPointer))
-      return false;
-   bool success = javaLangClassGetModifiersImpl(clazzPointer, modifiersForClass);
-   if (!success)
-      return false;
-   bool isModFlagSet = (modifiersForClass & J9AccEnum) ? true : false;
-   TR_OpaqueClassBlock *enumClass = getClassFromSignature("java/lang/Enum", 14, method);
-   TR_OpaqueClassBlock *superClass = getSuperClass(clazzPointer);
-   return (isModFlagSet && (enumClass == superClass));
+   return (TR::Compiler->cls.romClassOf(clazzPointer)->modifiers & J9AccEnum) ? true : false;
```

Comment:
Can you please add a comment here clarifying why this change in logic was made ?

Reply:
I will be reverting this. The previous implementation also checked that the class extends `java/lang/Enum`, without which `isEnum` should return false regardless of whether the `ENUM` flag is set or not.